ui/src/features/PageWrapper/WorkspaceId.tsx

```diff
@@ -30,7 +30,7 @@ const WorkspaceIdWrapper: React.FC<Props> = ({ children }) => {
 
   if (isLoading) return <Loading />;
 
-  if (!workspace)
+  if (!workspace || !currentWorkspace)
```

Comment:
I feel like this will show the NotFoundPage more often than before, since after isLoading becomes false, we will have workspace, but won't have currentWorkspace for another rerender.

Reply:
Hmm...you are right about that. What do you think could be the alternative? I need `currentWorkspace` down the children components and then I'll need to check and add the loading state there.